content/building-blocks/automate-and-execute-your-work/reproducible-work/google_cloud_docker.md

```diff
@@ -0,0 +1,155 @@
+
+{{% /tip %}}
+
+{{% codeblock %}}
+```bash
+# Build a Docker image and assign a name to it
+$ docker build -t your_image_name .
+```
+{{% /codeblock %}}
+
+### Step 3: Docker compose
+
+Docker compose is a tool within Docker whose main functionality is to coordinate your containers and provide them with the services required to run coherently. Although this can sound a bit confusing if you are new to Docker (You can learn more about Docker compose [here](https://docs.docker.com/compose/)), the key takeaway is that it will assist and simplify the task of replicating your environment by providing some extra information to Docker about how to do so.
+
+In the same way that you needed a dockerfile to provide Docker with the necessary instructions to build an image, a docker-compose file is required to tell docker how to provide these services to a (series of) container(s).
+
+{{% tip %}}
+
+Docker compose files must be YAML files, so make sure that your docker compose file is denoted as such with the extension ".yml" 
+
+{{% /tip %}}
+
+All Docker needs to carry out the instructions contained in our docker-compose file is to open the terminal at the location of our docker-compose file and excute the following code:
+
+{{% codeblock %}}
+```bash
+# Build the docker image
+$ docker compose up
+```
+{{% /codeblock %}}
+
+## Checking if the environment has been replicated successfully and accessing it.
+
+To check if docker has successfully replicated your environment you can go into your Docker desktop app and click on "containers" in the top left corner. If everything went correctly you should see the icons corresponding to the Docker compose instansce and the container containing your environment colored in green and their status set as "Running".
+
+<p align = "center">
+<img src = "../img/Check_status.png" width="750">
+<figcaption> This is what you should see on Docker desktop if the operation was successful. </figcaption>
+</p>
+
+If this is the case, congratulations! You have succefully replicated your environment with Docker! 
+
+### Option A: Jupyter Notebook
+
+Finally, to access to your environment click on the container and then click again on "logs" at the top right corner of your screen. On the last line of the output you are seeing now you will find a link that will take you directly to your environment in Jupyter.
+
+<p align = "center">
+<img src = "../img/Jupyter.png" width="750">
+<figcaption> Click on the link signaled to access Jupyter!</figcaption>
+</p>
+
+{{% tip %}}
+
+Instead of clicking on the link you can also directly copy it in your browser to enter the local port where the environment is located. If you do this do not forget to also copy the access token generated by Jupyter in order to access directly without needing to introduce it manually.

```

Comment:
This is kinda important if you want to access the notebook again and again without rebuilding the container. That token usually is not shown again if you restart an old container. So the trick is to copy that token and *make a new password* at the bottom of the Jupyter log-in page. Then if you re-start the notebook, you just log in with the password.

This becomes even more important when you move to remote systems (eg. Google cloud) and have to re-access notebooks by restarting containers.

Reply:
Perfect! The fragment will be expanded in that direction